### PR TITLE
[Storage Blobs Service] STG 74 features

### PR DESCRIPTION
**Pull request by Jinming-Hu**
closes https://github.com/Azure/azure-sdk-for-cpp/issues/474
closes https://github.com/Azure/azure-sdk-for-cpp/issues/477
closes https://github.com/Azure/azure-sdk-for-cpp/issues/665

related to https://github.com/Azure/azure-sdk-for-cpp/issues/478 https://github.com/Azure/azure-sdk-for-cpp/issues/660  https://github.com/Azure/azure-sdk-for-cpp/issues/659

### CODE REVIEW

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp
Comment (vhvb1989) [inline sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp:26]: why?

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp
Comment (vhvb1989) [inline sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp:1195]: since you have these strings ("NeverExpire", "RelativeToCreation", etc) in here and also for `ScheduleBlobExpiryOriginTypeToString`, can you add them to Details:: as constants and then just reuse them?

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp
Comment (vhvb1989) [inline sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp:5497]: Remove `unused()` function.
Update this to (void) options.
Think we agreed about this on some of the team meetings?

**Review by vhvb1989 — APPROVED** [review]
adding some questions/requests before merging.
Please address or comment back before merging

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp:26]: `2019-12-12` is STG73, `2020-02-10` is STG74.

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp:1195]: That's not easy for auto generator. This auto generator is supposed to be a temporary thing. So why bother wasting more efforts.

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp:5497]: I didn't remember we have such an agreement

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp:1195]: btw, core team is supposed to support to full-fledged auto generator in the near future. I think this is a very useful feature request.

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp
Comment (vhvb1989) [reply sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp:5497]: Didn't find the prev chat about it.  Started a new one here:

https://teams.microsoft.com/l/message/19:a5180a2de11a40cb83a822f588d60c15@thread.skype/1600274130341?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1600274130341&teamName=Azure%20SDK&channelName=Language%20-%20C%E1%90%A9%E1%90%A9&createdTime=1600274130341

Seems like `(void)arg` is what most of people like

**Review by katmsft — APPROVED** [review]

**Review by katmsft — APPROVED** [review]

**Pull request merged**
